src/python/py/models/builder.py

```diff
@@ -3297,6 +3299,195 @@ def make_rotary_embedding_caches(self, **kwargs):
+            self.attention_attrs["q_path"] = f"{q_rotary_name}/output_0"
+            k_rotary_name = f"/model/layers.{layer_id}/attn/k_rotary/RotaryEmbedding"
+            self.make_rotary_embedding(k_rotary_name, root_input=self.attention_attrs["k_path"], position_ids=kwargs.get("position_ids", "position_ids"))
+            self.attention_attrs["k_path"] = f"{k_rotary_name}/output_0"
+
+        # Make repeat KV nodes (Note: `repeat_kv` needs to be kept since GroupQueryAttention isn't supported for FP32 CUDA)
+        past_k = f"past_key_values.{layer_id}.key"
+        past_v = f"past_key_values.{layer_id}.value"
+        present_k = f"present.{layer_id}.key"
+        present_v = f"present.{layer_id}.value"
+        if self.num_attn_heads != self.num_kv_heads and self.attention_attrs["op_type"] == "MultiHeadAttention":
+            self.attention_attrs["k_path"] = self.make_repeat_kv(layer_id, root_input=self.attention_attrs["k_path"], past_kv=past_k, present_kv=present_k)
+            self.attention_attrs["v_path"] = self.make_repeat_kv(layer_id, root_input=self.attention_attrs["v_path"], past_kv=past_v, present_kv=present_v)
+            past_k, past_v, present_k, present_v = "", "", "", ""
+
+        # Make attention node (e.g. MultiHeadAttention, GroupQueryAttention, etc.)
+        attn_name = f"/model/layers.{layer_id}/attn/{self.attention_attrs['op_type']}"
+        self.make_attention_op(
+            attn_name, q_path=self.attention_attrs["q_path"], k_path=self.attention_attrs["k_path"], v_path=self.attention_attrs["v_path"],
+            past_k=past_k, past_v=past_v, present_k=present_k, present_v=present_v,
+            cos_cache=cos_cache_name, sin_cache=sin_cache_name, **kwargs,
+        )
+
+        # add an extra SimplifiedLayerNorm before the output projection for attention
+        attn_sub_norm_kwargs = {"epsilon": self.rms_norm_eps, "axis": -1, "stash_type": 1}   # BitNetRMSNorm on axes -1
+        attn_sub_norm_name = f"/model/layers.{layer_id}/attn/attn_sub_norm/SimplifiedLayerNormalization"
+        attn_sub_norm_weight_name = f"/model/layers.{layer_id}/attn/attn_sub_norm.weight"
+        attn_sub_norm_weight_name = attn_sub_norm_weight_name[1:].replace("/", ".")
+        attn_sub_norm_output = f"{attn_sub_norm_weight_name}/output_0"
+        self.make_external_tensor((attention.attn_sub_norm.weight.detach().to(self.to_torch_dtype[self.io_dtype]) + self.layernorm_attrs["add_offset"]).contiguous(), attn_sub_norm_weight_name)
+        self.make_node("SimplifiedLayerNormalization", inputs=[f"{attn_name}/output_0", attn_sub_norm_weight_name], outputs=[attn_sub_norm_output], name=attn_sub_norm_name, **attn_sub_norm_kwargs)
+
+        # Make MatMul node (output projection weight node)
+        o_proj = 'o_proj' if hasattr(attention, 'o_proj') else 'dense'
+        o_matmul_basename = f"/model/layers.{layer_id}/attn/o_proj/MatMul"
+        o_weight = eval(f"attention.{o_proj}")
```

Comment:
Avoid eval: it is a code smell

```suggestion
        o_weight = getattr(attention, o_proj)
```